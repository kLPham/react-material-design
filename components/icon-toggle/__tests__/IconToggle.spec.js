import React from 'react';
import { shallow } from 'enzyme';
import IconToggle from '../index';

describe('<IconToggle />', () => {
    it('should be able to render', () => {
        const iconToggle = shallow(
            <IconToggle />,
  );
        expect(iconToggle).toMatchSnapshot();
    });
    it('should render disabled', () => {
        const disabled = true;
        const iconToggle = shallow(
            <IconToggle disabled={disabled} />,
  );
        expect(iconToggle).toMatchSnapshot();
    });
    it('should render primary', () => {
        const primary = true;
        const iconToggle = shallow(
            <IconToggle primary={primary} />,
  );
        expect(iconToggle).toMatchSnapshot();
    });
    it('should render accent', () => {
        const accent = true;
        const iconToggle = shallow(
            <IconToggle accent={accent} />,
  );
        expect(iconToggle).toMatchSnapshot();
    });
    it('shound render with custom icons', () => {
        const icon = 'star_border';
        const toggledIcon = 'star';
        const iconToggle = shallow(
            <IconToggle icon={icon} toggledIcon={toggledIcon} />,
      );
        expect(iconToggle).toMatchSnapshot();
    });
});
