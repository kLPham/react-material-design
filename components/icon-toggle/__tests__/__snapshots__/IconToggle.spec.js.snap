// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IconToggle /> should be able to render 1`] = `
<i
  aria-label="Add to favorites"
  aria-pressed="false"
  className="mdc-icon-toggle material-icons"
  data-toggle-off="{\\"label\\": \\"Add to favorites\\", \\"content\\": \\"favorite_border\\"}"
  data-toggle-on="{\\"label\\": \\"Remove from favorites\\", \\"content\\": \\"favorite\\"}"
  role="button"
  tabIndex="0"
>
  favorite_border
</i>
`;

exports[`<IconToggle /> should render accent 1`] = `
<i
  aria-label="Add to favorites"
  aria-pressed="false"
  className="mdc-icon-toggle material-icons mdc-icon-toggle--accent"
  data-toggle-off="{\\"label\\": \\"Add to favorites\\", \\"content\\": \\"favorite_border\\"}"
  data-toggle-on="{\\"label\\": \\"Remove from favorites\\", \\"content\\": \\"favorite\\"}"
  role="button"
  tabIndex="0"
>
  favorite_border
</i>
`;

exports[`<IconToggle /> should render disabled 1`] = `
<i
  aria-label="Add to favorites"
  aria-pressed="false"
  className="mdc-icon-toggle material-icons"
  data-toggle-off="{\\"label\\": \\"Add to favorites\\", \\"content\\": \\"favorite_border\\"}"
  data-toggle-on="{\\"label\\": \\"Remove from favorites\\", \\"content\\": \\"favorite\\"}"
  role="button"
  tabIndex="0"
>
  favorite_border
</i>
`;

exports[`<IconToggle /> should render primary 1`] = `
<i
  aria-label="Add to favorites"
  aria-pressed="false"
  className="mdc-icon-toggle material-icons mdc-icon-toggle--primary"
  data-toggle-off="{\\"label\\": \\"Add to favorites\\", \\"content\\": \\"favorite_border\\"}"
  data-toggle-on="{\\"label\\": \\"Remove from favorites\\", \\"content\\": \\"favorite\\"}"
  role="button"
  tabIndex="0"
>
  favorite_border
</i>
`;

exports[`<IconToggle /> shound render with custom icons 1`] = `
<i
  aria-label="Add to favorites"
  aria-pressed="false"
  className="mdc-icon-toggle material-icons"
  data-toggle-off="{\\"label\\": \\"Add to favorites\\", \\"content\\": \\"star_border\\"}"
  data-toggle-on="{\\"label\\": \\"Remove from favorites\\", \\"content\\": \\"star\\"}"
  role="button"
  tabIndex="0"
>
  favorite_border
</i>
`;
